tf2_geometry_msgs/src/tf2_geometry_msgs/tf2_geometry_msgs.py

```diff
@@ -62,6 +63,89 @@ def transform_to_kdl(t):
                                     t.transform.translation.y,
                                     t.transform.translation.z))
 
+def transformCovariance(cov_in, transform):
+    # Converting the Quaternion to a Rotation Matrix first
+    # Taken from: https://automaticaddison.com/how-to-convert-a-quaternion-to-a-rotation-matrix/
+    q0 = transform.transform.rotation.w
+    q1 = transform.transform.rotation.x
+    q2 = transform.transform.rotation.y
+    q3 = transform.transform.rotation.z
+     
+    # First row of the rotation matrix
+    r00 = 2 * (q0 * q0 + q1 * q1) - 1
+    r01 = 2 * (q1 * q2 - q0 * q3)
+    r02 = 2 * (q1 * q3 + q0 * q2)
+     
+    # Second row of the rotation matrix
+    r10 = 2 * (q1 * q2 + q0 * q3)
+    r11 = 2 * (q0 * q0 + q2 * q2) - 1
+    r12 = 2 * (q2 * q3 - q0 * q1)
+     
+    # Third row of the rotation matrix
+    r20 = 2 * (q1 * q3 - q0 * q2)
+    r21 = 2 * (q2 * q3 + q0 * q1)
+    r22 = 2 * (q0 * q0 + q3 * q3) - 1
+
+    R =  np.array([[r00, r01, r02],
+                   [r10, r11, r12],
+                   [r20, r21, r22]])
+    
+    R_transpose = np.transpose(R)
+
+    cov_11 = np.array([cov_in[:3], cov_in[6:9], cov_in[12:15]])
+    cov_12 = np.array([cov_in[3:6], cov_in[9:12], cov_in[15:18]])
+    cov_21 = np.array([cov_in[18:21], cov_in[24:27], cov_in[30:33]])
+    cov_22 = np.array([cov_in[21:24], cov_in[27:30], cov_in[33:]])
+
+    # And we perform the transform
+    result_11 = R * cov_11 * R_transpose
+    result_12 = R * cov_12 * R_transpose
+    result_21 = R * cov_21 * R_transpose
+    result_22 = R * cov_22 * R_transpose
```

Comment:
For `np.array` types which represent 2D matrices, the `*` operator performs element-wise multiplication, and not the conventional matrix product that we're expecting here. We can get the desired multiplication by using `@` instead.

```suggestion
    result_11 = R @ cov_11 @ R_transpose
    result_12 = R @ cov_12 @ R_transpose
    result_21 = R @ cov_21 @ R_transpose
    result_22 = R @ cov_22 @ R_transpose
```

Reply:
Yes this was my mistake. Thanks for correcting, I will resolve this